cloud/app/server/utils.js

```diff
@@ -124,7 +140,372 @@ const waitForClickHouse = async () => {
+
+  await ClickHouse.client.exec({
+    query: `CREATE ROW POLICY IF NOT EXISTS default_admin ON default.* USING 1 TO ${process.env.CLICKHOUSE_USER || 'default'}`
+  });
+};
+
+/* creates ClickHouse user for an organization with SELECT access for all dbs and tables */
+const ensureClickHouseOrgUser = async (orgId) => {
+  const orgUser = `org_${orgId}_user`;
+  // Check if user exists
+  const userExists = await ClickHouse.client.query({
+    query: `SELECT name FROM system.users WHERE name = '${orgUser}'`,
+    format: 'JSONEachRow'
+  });
+  const accountsCollection = Mongo.db.collection('accounts');
+  const users = await userExists.json();
+  if (users.length > 0) {
+    log.debug(`ClickHouse user for organization ${orgId} already exists`);
+    const orgDoc = await accountsCollection.findOne({ _id: orgId });
+    const { user: orgUser, password } = orgDoc.clickhouseCredentials;
+    if (!password) {
+      throw new Error(`ClickHouse user for organization ${orgId} exists but no password found in mongo`);
+    } else {
+      log.debug(`retrieved ClickHouse credentials for organization ${orgId} from mongo: ${orgUser} / ${password}`);
+      return { 
+        user: orgUser,
+        password: password
+      };
+    }
+  }
+
+  const orgPassword = getRandomId(15);
+
+  log.debug(`creating ClickHouse user for organization ${orgId} : ${orgUser} / ${orgPassword}`);
+
+  // create user
+  await ClickHouse.client.exec({
+    query: `CREATE USER IF NOT EXISTS ${orgUser} IDENTIFIED WITH plaintext_password BY '${orgPassword}'`
+  });
+
+  // grant read only access to all databases and tables - row level security will limit access to own org data
+  await ClickHouse.client.exec({
+    query: `GRANT SELECT ON *.* TO ${orgUser}`
+  });
+
+  // store user and password in mongo
+  await accountsCollection.updateOne(
+    { _id: orgId },
+    { $set: { clickhouseCredentials: { user: orgUser, password: orgPassword } } }
+  );
+
+  return {
+    user: orgUser,
+    password: orgPassword
+  };
+}
+
+/** Create HyperDX team document */
+const createHyperDXTeam = (teamId, teamName) => {
+  const now = new Date();
+  return {
+    _id: teamId,
+    name: teamName,
+    allowedAuthMethods: [],
+    collectorAuthenticationEnforced: true,
+    hookId: getRandomId(36),
+    apiKey: getRandomId(36),
+    createdAt: now,
+    updatedAt: now,
+    __v: 0
+  };
+};
+
+/** Create HyperDX user document */
+const createHyperDXUser = (userId, email, password, teamId) => {
+  const now = new Date();
+  const salt = crypto.randomBytes(32).toString('hex');
+  const hash = crypto.pbkdf2Sync(password, salt, 25000, 512, 'sha256').toString('hex');
+  
+  return {
+    _id: userId,
+    email,
+    name: email,
+    accessKey: getRandomId(36),
+    salt,
+    hash,
+    team: teamId,
+    createdAt: now,
+    updatedAt: now,
+    __v: 0
+  };
+};
+
+/** Create HyperDX connection document */
+const createHyperDXConnection = (connectionId, teamId, name, username, password) => {
```

Comment:
oh, come on! A function to do nothing but create an object and that function is only ever used once!? This is the kind of AI slop that makes reviewing AI code take more time then writing it oneself.